### ISSUE TITLE
Stable release

### ISSUE THREAD

**Issue by theofidry** [bamarni/composer-bin-plugin#10]
Is there anything left before having a stable release (or beta/rc)?

Also is there any point in keeping all the classes as extendable i.e. not making them final? (that is actually the only last change I would like to submit)

**Comment by bamarni** [comment]
cf. #12, I didn't got that much feedback except from you :D but I think it's stable. At the moment the main interesting part missing would be some integration or something similar to the merge plugin, to avoid incompatibilities between project deps and bin deps.

I could tag a v1 after #11 is there probably. As per marking the classes final yes, why not.

**Comment by theofidry** [comment]
>At the moment the main interesting part missing would be some integration or something similar to the merge plugin, to avoid incompatibilities between project deps and bin deps.

Well actually that's what I'm working on :P https://github.com/theofidry/composer-inheritance-plugin

To recap, for using *phars*, I'm using https://github.com/tommy-muehle/tooly-composer-script. Easy and simple. However phars have two limitations: 1. no versioning (which can be and is ok most of the time) 2. cannot execute code (otherwise you may have silent dep conflicts).

So as described in https://github.com/bamarni/composer-bin-plugin/issues/12, the main usage I have for this plugin is phars for which I need a versioning or framework bridges. In the later case, as we already discussed somewhere, as you child `composer.json` does not "inherit" from the parent one, there may still be dep conflicts besides you have to do two autoloadings: the one from the root vendor and the one from the child. The plugin I'm working on just wraps https://github.com/wikimedia/composer-merge-plugin to avoid any weird configuration for this specific usage, install it in the child `composer.json` and:

- The `autoload`, `autoload-dev`, `require` and `require-dev` sections from the root `composer.json` will be merge with the children. As a result, you have no silent dep conflicts and you have only the `autoload.php` of the child to require.

On my end, I only need #11 before having everything I need from this plugin. But yeah it would be nice to have other people feedback as well :P
